File: src/main/java/de/churl/cookbook/model/Recipe.java
package de.churl.cookbook.model;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import javax.persistence.Table;
import javax.validation.constraints.Size;
import java.util.Set;
import java.util.UUID;

@Entity
@Table(name = "recipes")
@Getter
@Setter
@NoArgsConstructor
public class Recipe {

    @Id
    @Column(name = "recipe_id")
    private UUID recipeID;

    @Size(max = 127)
    @Column(name = "recipe_title", length = 127)
    private String recipeTitle;

    @Size(max = 255)
    @Column(name = "recipe_descr")
    private String recipeDescr;

    @Size(max = 8191)
    @Column(name = "recipe_body", length = 8191)
    private String recipeBody;

    @OneToMany(mappedBy = "recipe", cascade = CascadeType.ALL, orphanRemoval = true)
    Set<IngredientUsage> ingredientUsages;
}

File: src/main/kotlin/de/churl/cookbook/model/Recipe.kt
package de.churl.cookbook.model

import de.churl.cookbook.util.emptyUUID
import org.hibernate.annotations.GenericGenerator
import java.util.*
import javax.persistence.*
import javax.validation.constraints.Size

@Entity
@Table(name = "recipes")
data class Recipe(
    @Size(max = 127)
    @Column(name = "recipe_title", length = 127)
    var recipeTitle: String,

    @Size(max = 255)
    @Column(name = "recipe_descr", length = 255)
    var recipeDescr: String,

    @Size(max = 8191)
    @Column(name = "recipe_body", length = 8191)
    var recipeBody: String,

    @OneToMany(mappedBy = "recipe", cascade = [CascadeType.ALL], orphanRemoval = true)
    var ingrUsages: MutableCollection<IngrUsage> = mutableSetOf()
) {
    // Attributes inside the class-body are not used in method derivations (i.e. hashCode)
    @Id
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(
        name = "UUID",
        strategy = "org.hibernate.id.UUIDGenerator",
    )
    @Column(name = "recipe_id")
    var recipeID: UUID = emptyUUID()
}
